src/Cons.js

```diff
@@ -0,0 +1,64 @@
+const
+  R = require('ramda'),
+  RF = require('ramda-fantasy'),
+  Fold = require('./Fold'),
+  Lens = require('./Lens'),
+  Prism = require('./Prism'),
+  Category = require('./Internal/Category'),
+
+  Either = RF.Either,
+  Tuple = RF.Tuple;
+
+
+//:: Prism [a] [b] (a, [a]) (b, [b])
+const _Cons = Prism.prism(
+  t => R.prepend(Tuple.fst(t), Tuple.snd(t)),
+  R.ifElse(R.isEmpty,
+           Either.Left,
+           xss => Either.Right(Tuple(R.head(xss), R.tail(xss)))));
+
+//:: Prism [a] [b] ([a], a) ([b], b)
+const _Snoc = Prism.prism(
+  t => R.append(Tuple.snd(t), Tuple.fst(t)),
+  R.ifElse(R.isEmpty,
+           Either.Left,
+           xss => Either.Right(Tuple(R.init(xss), R.last(xss)))));
+
+//:: a -> [a] -> [a]
+const cons = R.curry((x, xs) =>
+  Prism.review(_Cons, Tuple(x, xs)));
+
+//:: [a] -> Maybe (Tuple a [a])
+const uncons = Fold.preview(_Cons);
```

Comment:
Any objection to including the name in the type signature?

```javascript
//    uncons :: [a] -> Maybe (Tuple a [a])
const uncons = Fold.preview(_Cons);
```

I like to be able to find functions by running `ack ' <name> :: '`. :)